Update FileSystem::writeToFile() to take in a std::span

cdumez opened the pull request:
#### e173385dac3703f5d92018a5f4cc1619bf00cefc
<pre>
Update FileSystem::writeToFile() to take in a std::span
<a href="https://bugs.webkit.org/show_bug.cgi?id=274333">https://bugs.webkit.org/show_bug.cgi?id=274333</a>

Reviewed by Darin Adler.

* Source/JavaScriptCore/API/JSScript.mm:
(-[JSScript writeCache:]):
* Source/JavaScriptCore/jsc.cpp:
(JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/runtime/CachePayload.h:
(JSC::CachePayload::span const):
* Source/JavaScriptCore/runtime/CachedBytecode.cpp:
(JSC::CachedBytecode::commitUpdates const):
* Source/JavaScriptCore/runtime/CachedBytecode.h:
* Source/JavaScriptCore/runtime/CachedTypes.cpp:
(JSC::Encoder::releaseMapped):
(JSC::Encoder::Page::mutableSpan):
(JSC::Encoder::Page::span const):
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/wasm/WasmStreamingParser.cpp:
(JSC::Wasm::dumpWasmSource):
* Source/WTF/wtf/FileSystem.cpp:
(WTF::FileSystemImpl::appendFileContentsToFileHandle):
(WTF::FileSystemImpl::readOrMakeSalt):
(WTF::FileSystemImpl::overwriteEntireFile):
* Source/WTF/wtf/FileSystem.h:
* Source/WTF/wtf/posix/FileSystemPOSIX.cpp:
(WTF::FileSystemImpl::writeToFile):
* Source/WTF/wtf/win/FileSystemWin.cpp:
(WTF::FileSystemImpl::writeToFile):
* Source/WebCore/Modules/filesystemaccess/FileSystemSyncAccessHandle.cpp:
(WebCore::FileSystemSyncAccessHandle::write):
* Source/WebCore/Modules/model-element/scenekit/SceneKitModelLoaderUSD.mm:
(WebCore::writeToTemporaryFile):
* Source/WebCore/bindings/js/GCController.cpp:
(WebCore::GCController::dumpHeapForVM):
* Source/WebCore/contentextensions/SerializedNFA.cpp:
(WebCore::ContentExtensions::writeAllToFile):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::ApplicationCacheStorage::writeDataToUniqueFileInDirectory):
* Source/WebCore/platform/FileHandle.cpp:
(WebCore::FileHandle::write):
(WebCore::FileHandle::printf):
* Source/WebCore/platform/FileHandle.h:
* Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp:
(WebCore::transcodeImage):
* Source/WebCore/platform/network/BlobRegistryImpl.cpp:
(WebCore::writeFilePathsOrDataBuffersToFile):
* Source/WebCore/storage/StorageUtilities.cpp:
(WebCore::StorageUtilities::writeOriginToFile):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::createTemporaryFile):
* Source/WebCore/workers/service/server/SWScriptStorage.cpp:
(WebCore::SWScriptStorage::store):
* Source/WebKit/NetworkProcess/NetworkDataTaskBlob.cpp:
(WebKit::NetworkDataTaskBlob::writeDownload):
* Source/WebKit/NetworkProcess/NetworkDataTaskDataURL.cpp:
(WebKit::NetworkDataTaskDataURL::downloadDecodedData):
* Source/WebKit/NetworkProcess/ServiceWorker/ServiceWorkerDownloadTask.cpp:
(WebKit::ServiceWorkerDownloadTask::didReceiveData):
* Source/WebKit/NetworkProcess/cache/NetworkCache.cpp:
(WebKit::NetworkCache::Cache::dumpContentsToFile):
* Source/WebKit/NetworkProcess/cache/NetworkCacheIOChannelCurl.cpp:
(WebKit::NetworkCache::IOChannel::write):
* Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp:
(WebKit::NetworkDataTaskCurl::curlDidReceiveData):
* Source/WebKit/NetworkProcess/storage/BackgroundFetchStoreManager.cpp:
(WebKit::BackgroundFetchStoreManager::storeFetchResponseBodyChunk):
* Source/WebKit/Shared/PersistencyUtils.cpp:
(WebKit::writeToDisk):
* Source/WebKit/Shared/WebMemorySampler.cpp:
(WebKit::WebMemorySampler::writeHeaders):
(WebKit::WebMemorySampler::appendCurrentMemoryUsageToFile):
* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::writeSandboxDataToCacheFile):
(WebKit::compileAndCacheSandboxProfile):
* Source/WebKit/UIProcess/API/APIContentRuleListStore.cpp:
(API::writeDataToFile):
(API::compiledToFile):
(API::ContentRuleListStore::invalidateContentRuleListVersion):
* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/UIProcess/Inspector/win/WebInspectorUIProxyWin.cpp:
(WebKit::WebInspectorUIProxy::showSavePanelForSingleFile):
* Source/WebKit/UIProcess/WebFullScreenManagerProxy.cpp:
(WebKit::WebFullScreenManagerProxy::prepareQuickLookImageURL const):
* Source/WebKit/UIProcess/ios/WKModelView.mm:
(-[WKModelView createFileForModel:]):
* Source/WebKit/WebProcess/Model/mac/ARKitInlinePreviewModelPlayerMac.mm:
(WebKit::ARKitInlinePreviewModelPlayerMac::createFile):
* Tools/TestWebKitAPI/Tests/WTF/FileSystem.cpp:
(TestWebKitAPI::createTestFile):
(TestWebKitAPI::FileSystemTest::tempFilePath const): Deleted.
(TestWebKitAPI::FileSystemTest::tempFileSymlinkPath const): Deleted.
(TestWebKitAPI::FileSystemTest::tempEmptyFolderPath const): Deleted.
(TestWebKitAPI::FileSystemTest::tempEmptyFolderSymlinkPath const): Deleted.
(TestWebKitAPI::FileSystemTest::tempEmptyFilePath const): Deleted.
(TestWebKitAPI::FileSystemTest::spaceContainingFilePath const): Deleted.
(TestWebKitAPI::FileSystemTest::bangContainingFilePath const): Deleted.
(TestWebKitAPI::FileSystemTest::quoteContainingFilePath const): Deleted.
(TestWebKitAPI::TEST_F): Deleted.
(TestWebKitAPI::runGetFileModificationTimeTest): Deleted.
* Tools/TestWebKitAPI/Tests/WebCore/FileMonitor.cpp:
(TestWebKitAPI::TEST_F):
* Tools/TestWebKitAPI/Tests/WebCore/SharedBufferTest.cpp:
(TestWebKitAPI::FragmentedSharedBufferTest::SetUp):
* Tools/TestWebKitAPI/Tests/WebCore/SharedBufferTest.h:
(TestWebKitAPI::FragmentedSharedBufferTest::testData):

Canonical link: <a href="https://commits.webkit.org/278941@main">https://commits.webkit.org/278941@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c5085feae2256cc25fdd4d9308d907483250d698

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52035 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31358 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4440 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55308 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/2747 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/54337 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/37739 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2453 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/42363 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/1753 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54129 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28980 "1 flakes") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44884 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23432 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/26263 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2167 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/917 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/45380 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/48162 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2306 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56899 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/51542 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27150 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/2284 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/49754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/28388 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45003 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/48967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/29290 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/63849 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7605 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28128 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12054 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebKit/NetworkProcess/cache/NetworkCache.cpp:
```diff
@@ -638,8 +638,8 @@ void Cache::dumpContentsToFile()
 
-    static const char prologue[] = "{\n\"entries\": [\n";
-    writeToFile(fd, prologue, strlen(prologue));
+    static constexpr auto prologue = "{\n\"entries\": [\n"_s;
```
I don’t think we need `static`.

darinadler commented on Tools/TestWebKitAPI/Tests/WTF/FileSystem.cpp:
```diff
@@ -34,13 +34,13 @@
 namespace TestWebKitAPI {
 
-const char* FileSystemTestData = "This is a test";
+static constexpr auto FileSystemTestData = "This is a test"_s;
```
I would not include `static` here.

cdumez committed:
Update FileSystem::writeToFile() to take in a std::span
https://bugs.webkit.org/show_bug.cgi?id=274333

Reviewed by Darin Adler.

* Source/JavaScriptCore/API/JSScript.mm:
(-[JSScript writeCache:]):
* Source/JavaScriptCore/jsc.cpp:
(JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/runtime/CachePayload.h:
(JSC::CachePayload::span const):
* Source/JavaScriptCore/runtime/CachedBytecode.cpp:
(JSC::CachedBytecode::commitUpdates const):
* Source/JavaScriptCore/runtime/CachedBytecode.h:
* Source/JavaScriptCore/runtime/CachedTypes.cpp:
(JSC::Encoder::releaseMapped):
(JSC::Encoder::Page::mutableSpan):
(JSC::Encoder::Page::span const):
* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/wasm/WasmStreamingParser.cpp:
(JSC::Wasm::dumpWasmSource):
* Source/WTF/wtf/FileSystem.cpp:
(WTF::FileSystemImpl::appendFileContentsToFileHandle):
(WTF::FileSystemImpl::readOrMakeSalt):
(WTF::FileSystemImpl::overwriteEntireFile):
* Source/WTF/wtf/FileSystem.h:
* Source/WTF/wtf/posix/FileSystemPOSIX.cpp:
(WTF::FileSystemImpl::writeToFile):
* Source/WTF/wtf/win/FileSystemWin.cpp:
(WTF::FileSystemImpl::writeToFile):
* Source/WebCore/Modules/filesystemaccess/FileSystemSyncAccessHandle.cpp:
(WebCore::FileSystemSyncAccessHandle::write):
* Source/WebCore/Modules/model-element/scenekit/SceneKitModelLoaderUSD.mm:
(WebCore::writeToTemporaryFile):
* Source/WebCore/bindings/js/GCController.cpp:
(WebCore::GCController::dumpHeapForVM):
* Source/WebCore/contentextensions/SerializedNFA.cpp:
(WebCore::ContentExtensions::writeAllToFile):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::ApplicationCacheStorage::writeDataToUniqueFileInDirectory):
* Source/WebCore/platform/FileHandle.cpp:
(WebCore::FileHandle::write):
(WebCore::FileHandle::printf):
* Source/WebCore/platform/FileHandle.h:
* Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp:
(WebCore::transcodeImage):
* Source/WebCore/platform/network/BlobRegistryImpl.cpp:
(WebCore::writeFilePathsOrDataBuffersToFile):
* Source/WebCore/storage/StorageUtilities.cpp:
(WebCore::StorageUtilities::writeOriginToFile):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::createTemporaryFile):
* Source/WebCore/workers/service/server/SWScriptStorage.cpp:
(WebCore::SWScriptStorage::store):
* Source/WebKit/NetworkProcess/NetworkDataTaskBlob.cpp:
(WebKit::NetworkDataTaskBlob::writeDownload):
* Source/WebKit/NetworkProcess/NetworkDataTaskDataURL.cpp:
(WebKit::NetworkDataTaskDataURL::downloadDecodedData):
* Source/WebKit/NetworkProcess/ServiceWorker/ServiceWorkerDownloadTask.cpp:
(WebKit::ServiceWorkerDownloadTask::didReceiveData):
* Source/WebKit/NetworkProcess/cache/NetworkCache.cpp:
(WebKit::NetworkCache::Cache::dumpContentsToFile):
* Source/WebKit/NetworkProcess/cache/NetworkCacheIOChannelCurl.cpp:
(WebKit::NetworkCache::IOChannel::write):
* Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp:
(WebKit::NetworkDataTaskCurl::curlDidReceiveData):
* Source/WebKit/NetworkProcess/storage/BackgroundFetchStoreManager.cpp:
(WebKit::BackgroundFetchStoreManager::storeFetchResponseBodyChunk):
* Source/WebKit/Shared/PersistencyUtils.cpp:
(WebKit::writeToDisk):
* Source/WebKit/Shared/WebMemorySampler.cpp:
(WebKit::WebMemorySampler::writeHeaders):
(WebKit::WebMemorySampler::appendCurrentMemoryUsageToFile):
* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::writeSandboxDataToCacheFile):
(WebKit::compileAndCacheSandboxProfile):
* Source/WebKit/UIProcess/API/APIContentRuleListStore.cpp:
(API::writeDataToFile):
(API::compiledToFile):
(API::ContentRuleListStore::invalidateContentRuleListVersion):
* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/UIProcess/Inspector/win/WebInspectorUIProxyWin.cpp:
(WebKit::WebInspectorUIProxy::showSavePanelForSingleFile):
* Source/WebKit/UIProcess/WebFullScreenManagerProxy.cpp:
(WebKit::WebFullScreenManagerProxy::prepareQuickLookImageURL const):
* Source/WebKit/UIProcess/ios/WKModelView.mm:
(-[WKModelView createFileForModel:]):
* Source/WebKit/WebProcess/Model/mac/ARKitInlinePreviewModelPlayerMac.mm:
(WebKit::ARKitInlinePreviewModelPlayerMac::createFile):
* Tools/TestWebKitAPI/Tests/WTF/FileSystem.cpp:
(TestWebKitAPI::createTestFile):
(TestWebKitAPI::FileSystemTest::tempFilePath const): Deleted.
(TestWebKitAPI::FileSystemTest::tempFileSymlinkPath const): Deleted.
(TestWebKitAPI::FileSystemTest::tempEmptyFolderPath const): Deleted.
(TestWebKitAPI::FileSystemTest::tempEmptyFolderSymlinkPath const): Deleted.
(TestWebKitAPI::FileSystemTest::tempEmptyFilePath const): Deleted.
(TestWebKitAPI::FileSystemTest::spaceContainingFilePath const): Deleted.
(TestWebKitAPI::FileSystemTest::bangContainingFilePath const): Deleted.
(TestWebKitAPI::FileSystemTest::quoteContainingFilePath const): Deleted.
(TestWebKitAPI::TEST_F): Deleted.
(TestWebKitAPI::runGetFileModificationTimeTest): Deleted.
* Tools/TestWebKitAPI/Tests/WebCore/FileMonitor.cpp:
(TestWebKitAPI::TEST_F):
* Tools/TestWebKitAPI/Tests/WebCore/SharedBufferTest.cpp:
(TestWebKitAPI::FragmentedSharedBufferTest::SetUp):
* Tools/TestWebKitAPI/Tests/WebCore/SharedBufferTest.h:
(TestWebKitAPI::FragmentedSharedBufferTest::testData):

Canonical link: https://commits.webkit.org/278941@main
